odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/DungeonWaypoints.kt

```diff
@@ -64,21 +74,92 @@ object DungeonWaypoints : Module(
         val filled: Boolean,
         val depth: Boolean,
         val aabb: AxisAlignedBB,
-        val title: String?
+        val title: String?,
+        val secret: Boolean,
+        var clicked: Boolean = false
     )
 
     override fun onKeybind() {
         allowEdits = !allowEdits
         modMessage("Dungeon Waypoint editing ${if (allowEdits) "§aenabled" else "§cdisabled"}§r!")
     }
 
+    private val secretEntities = mutableMapOf<Int, Vec3>()
+
+    private val drops = listOf(
+        "Health Potion VIII Splash Potion", "Healing Potion 8 Splash Potion", "Healing Potion VIII Splash Potion",
+        "Decoy", "Inflatable Jerry", "Spirit Leap", "Trap", "Training Weights", "Defuse Kit", "Dungeon Chest Key", "Treasure Talisman", "Revive Stone",
+    )
+
+    fun resetSecrets() {
+        val room = DungeonUtils.currentRoom
+        for ((_, waypointsList) in DungeonWaypointConfigCLAY.waypoints.filter { waypoints -> waypoints.value.any { it.clicked } }) {
+            waypointsList.filter { it.clicked }.forEach { it.clicked = false }
+        }
+
+        if (room != null) DungeonUtils.setWaypoints(room)
+        glList = -1
+        secretEntities.clear()
+    }
+
+    init {
+        onWorldLoad {
+            resetSecrets()
+        }
+
+        onMessage(Regex("(?s).*(\\d+)/\\1 Secrets.*")) {
```

Comment:
What is this?

Reply:
detecting when a room is full cleared to clear secrets, for example if someone else got all the secrets you wouldnt still need to know them